core/src/main/java/io/grpc/Metadata.java

```diff
@@ -362,17 +364,30 @@ private String toStringInternal() {
    *
    * <p>Only the following ASCII characters are allowed in the names of keys:
    * <ul>
-   *   <li>letters, i.e., {@code a-z} and {@code A-Z}</li>
-   *   <li>digits, i.e., {@code 0-9}</li>
-   *   <li>dash, i.e., {@code -}</li>
-   *   <li>underscore, i.e., {@code _}</li>
+   *   <li>letters: {@code a-z}</li>
```

Comment:
Not sure I like the idea of maintaining 2 strings. If we're going to force things to lowercase, maybe we just say so?